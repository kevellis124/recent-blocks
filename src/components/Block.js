import React, { Component } from "react";
import PropTypes from "prop-types";

class Block extends Component {
  constructor() {
    super();
    this.state = { showDetails: false };
  }

  handleClick = () => {
    this.setState({ showDetails: !this.state.showDetails });
  };

  render() {
    return (
      <div className={"block"} onClick={this.handleClick}>
        <ul>Block id: {this.props.id}</ul>
        <ul>Block timestamp: {this.props.timestamp}</ul>
        <ul>Block action count: {this.props.actionCount}</ul>
        {this.state.showDetails && (
          <ul>Raw block dump: {JSON.stringify(this.props.rawBlock)}</ul>
        )}
        <br/>
      </div>
    );
  }
}
Block.propTypes = {
  id: PropTypes.string,
  timestamp: PropTypes.string,
  actionCount: PropTypes.number,
  rawBlock: PropTypes.object
};

Block.defaultProps = {
  id: "default id",
  timestamp: "12:00:00",
  actionCount: 0,
  rawBlock: {}
};

export default Block;
