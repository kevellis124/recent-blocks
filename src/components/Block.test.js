import { shallow } from "../enzyme";
import * as React from "react";
import Block from "./Block";

describe("<Block />", () => {
    describe("By default", () => {
        let component;
        beforeAll(() => {
            component = shallow(<Block />);
        });
        it("hasnt changed", () => {
            expect(component.html()).toMatchSnapshot();
        });
    });
});
