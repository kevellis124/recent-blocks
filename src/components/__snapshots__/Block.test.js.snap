// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> By default hasnt changed 1`] = `"<div class=\\"block\\"><ul>Block id: default id</ul><ul>Block timestamp: 12:00:00</ul><ul>Block action count: 0</ul><br/></div>"`;
